js/customize-snapshots.js

```diff
@@ -731,16 +701,69 @@
 		 * @returns {string} URL.
 		 */
 		getSnapshotFrontendPreviewUrl: function getSnapshotFrontendPreviewUrl() {
-			var snapshot = this, a = document.createElement( 'a' );
+			var snapshot = this, a = document.createElement( 'a' ),
+				params = {};
+
+			if ( api.settings.changeset && api.settings.changeset.uuid ) {
+				params.customize_changeset_uuid = api.settings.changeset.uuid;
+			} else {
+				params.customize_changeset_uuid = snapshot.data.uuid;
+			}
+
 			a.href = snapshot.frontendPreviewUrl.get();
-			if ( a.search ) {
-				a.search += '&';
+			if ( snapshot.statusButton.disbleButton.get() ) {
+				return a.href;
+			}
+			if ( ! api.settings.theme.active ) {
+				params.theme = api.settings.theme.stylesheet;
 			}
-			a.search += snapshot.uuidParam + '=' + snapshot.data.uuid;
+			a.search = $.param( params );
+
 			return a.href;
 		},
 
 		/**
+		 * Frontend preview link setup.
+		 *
+		 * @returns {void}
+		 */
+		setUpPreviewLink: function setUpPreviewLink() {
+			var snapshot = this;
+
+			snapshot.previewLink = $( $.trim( wp.template( 'snapshot-preview-link' )() ) );
+			snapshot.previewLink.attr( 'target', snapshot.data.uuid );
+			snapshot.previewLink.href = snapshot.frontendPreviewUrl.get();
+
+			snapshot.previewLink.click( function( e ) {
+				var onceProcessingComplete;
+				e.preventDefault();
+
+				snapshot.previewLink.href = snapshot.getSnapshotFrontendPreviewUrl();
```

Comment:
@miina As I was testing this, there was one use case that I found that I missed with the old version. Namely, I would often right-click on the frontend preview link in order to obtain the URL to then share with someone else or to open it in another browser. In the way that this is implemented now, this does not work. I think it should restore the previous behavior whereby there was a `setPreviewLinkHref` function that would be responsible for keeping this link up-to-date always. Then, sure, there can still be a `click` handler which is responsible for ensuring that we `requestChangesetUpdate`. But then this `requestChangesetUpdate` call could be done on `mousedown` to also capture the case where a user is copying the URL. Some combination of those events I think are needed to get it so that we can request the update both for users who try to copy the URL and for others who just want to do a simple click on the link.